deepmd/pt/model/network/layernorm.py

```diff
@@ -0,0 +1,126 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import torch
+import torch.nn as nn
+
+from deepmd.dpmodel.utils.network import LayerNorm as DPLayerNorm
+from deepmd.pt.utils import (
+    env,
+)
+from deepmd.pt.utils.env import (
+    DEFAULT_PRECISION,
+    PRECISION_DICT,
+)
+from deepmd.pt.utils.utils import (
+    to_numpy_array,
+    to_torch_tensor,
+)
+
+from .mlp import (
+    MLPLayer,
+)
+
+device = env.DEVICE
+
+
+class LayerNorm(MLPLayer):
+    def __init__(
+        self,
+        num_in,
+        eps: float = 1e-5,
+        uni_init: bool = True,
+        bavg: float = 0.0,
+        stddev: float = 1.0,
+        precision: str = DEFAULT_PRECISION,
+    ):
+        self.eps = eps
+        self.uni_init = uni_init
+        self.num_in = num_in
+        super().__init__(
+            num_in=1,
+            num_out=num_in,
+            bias=True,
+            use_timestep=False,
+            activation_function=None,
+            resnet=False,
+            bavg=bavg,
+            stddev=stddev,
+            precision=precision,
+        )
+        self.matrix = torch.nn.Parameter(self.matrix.squeeze(0))
+        if self.uni_init:
+            nn.init.ones_(self.matrix.data)
+            nn.init.zeros_(self.bias.data)
+
+    def dim_out(self) -> int:
+        return self.matrix.shape[0]
+
+    def forward(
+        self,
+        xx: torch.Tensor,
+    ) -> torch.Tensor:
+        """One Layer Norm used by DP model.
+
+        Parameters
+        ----------
+        xx : torch.Tensor
+            The input of index.
+
+        Returns
+        -------
+        yy: torch.Tensor
+            The output.
+        """
+        mean = xx.mean(dim=-1, keepdim=True)
+        variance = xx.var(dim=-1, unbiased=False, keepdim=True)
+        yy = (xx - mean) / torch.sqrt(variance + self.eps)
+        if self.matrix is not None:
+            yy = yy * self.matrix
+        if self.bias is not None:
+            yy = yy + self.bias
```

Comment:
it seems that the `self.matrix` and `self.bias` are not updated at all. If it is the situation, then they should be removed.